### DISCUSSION TITLE
Certificates rotation

### DISCUSSION THREAD

**Discussion by uk1988** [rancher/rke2#3978] [Q&A] [answered]
Hi all

We have a 3-node rke2 cluster where each node is master and agent.
On how to rotate client certificates we have found the doc: https://docs.rke2.io/advanced

In addition, I have some questions:

Do we need to run the certificate rotation on every node? 

The command only rotates all of the client certificates that are located in /var/lib/rancher/rke2/server/tls.
How do we rotate the server certificates? I know they are valid for 10 years but still how do we rotate them?
Do we need to do it manually with openssl? 

Did not try this yet but what happens to the certificates when you perform a k8s upgrade with rke2?
I assume the client certificate gets updated during the upgrade. However what does happen with the server certificates? 

And last I have a recommendation. It would be nice to have a command that would output validity of all deployed certificates.
Similar as what is possible in kubeadmin.

**Comment by brandond** [comment, marked as answer]
CA cert rotation and custom CA certs are not supported in any available release. You can see the proposal at https://github.com/k3s-io/k3s/blob/master/docs/adrs/ca-cert-rotation.md - the initial implementation of this should be in the next set of releases, and docs are currently work in progress.

**Reply by brandond** [reply]
There is not much of a need to check the validity, as the certs are regenerated on startup if necessary. If there is any question about validity or approaching expiration, just restart the rke2 service.

**Reply by uk1988** [reply]
Good to see that there is a proposal on CA cert rotation. 
Until that is not in place can we rotate this manually with OpenSSL?
In vanilla k8s you can put temp 2 certs as CA and then phase out the old one. 

What does happen when you upgrade the k8s with rke2? I assume the client certificate gets updated during the upgrade but the server does not.

The reason for me being pushy on the subject is that we will install a small on-prem k8s system and live them for some time.

**Reply by brandond** [reply]
> What does happen when you upgrade the k8s with rke2? I assume the client certificate gets updated during the upgrade but the server does not.

I'm not sure I'm understanding the question. Why would client certificates need updating when the version changes?

**Reply by uk1988** [reply]
I was just wondering if that is the case. Thank you for your answers.
